File: src/main/java/org/reactome/server/tools/Checker.java
package org.reactome.server.tools;

import java.time.Duration;
import java.time.Instant;

public interface Checker extends Runnable {
    @Override
    default void run() {
        Instant start = Instant.now();
        process();
        System.out.println(this.getClass().getSimpleName() + " finished in " + Duration.between(start, Instant.now()).toString().substring(2));
    }

    void process();

    int getFailedChecks();

    int getTotalChecks();
}

File: src/main/java/org/reactome/server/tools/Checker.kt
package org.reactome.server.tools

import java.time.Duration
import java.time.Instant

interface Checker : Runnable {
    override fun run() {
        val start = Instant.now()
        process()
        println("${javaClass.simpleName} finished in ${Duration.between(start, Instant.now()).toString().substring(2)}")
    }

    fun process()

    fun getFailedChecks(): Int
    fun getTotalChecks(): Int
}
